packages/x-charts/src/internals/defaultizeColorPerSeries.ts

```diff
@@ -0,0 +1,10 @@
+import { AllSeriesType } from '../models/seriesType';
+import { ChartSeriesType } from '../models/seriesType/config';
+
+export function defaultizeColorPerSeries<T extends ChartSeriesType>(
+  series: AllSeriesType<T>,
+  seriesIndex: number,
+  colors: string[],
+) {
+  return { color: colors[seriesIndex % colors.length], ...series };
+}
```

Comment:
This function feels a bit useless 🤔 

I know it is used in more places, but we could simply move this logic to those places instead as it is quite simple, and would probably skip one destructuring 😆